### PR TITLE
Major Refactoring and Rewrite of PyTorch Code

### PR DESCRIPTION
**Pull request by sauhaardac**
Rewrote the code base to follow official Python PEP8 Standards and use a OOP style rather than the previous dictionary approach. Please review carefully before accepting into the repository.

### CODE REVIEW

**Comment by sauhaardac** [comment]
In kzpy3.2/pytorch3/Utils.py:save_net, there is no code for saving the loss record, but the loss record is passed to the function. There is also code in Train.py for restoring saves for loss records. I kept this code how it is, with the restore code intact and no saving code. I also tried searching in bdd3 for a save file for the loss record, but couldn't find anything. Should I implement this saving feature, and if so what would the structure of the save look like?

**Comment by sauhaardac** [comment]
In kzpy3.2/pytorch3/Utils.py:save_net, there is no code for saving the loss record, but the loss record is passed to the function. There is also code in Train.py for restoring saves for loss records. I kept this code how it is, with the restore code intact and no saving code. I also tried searching in bdd3 for a save file for the loss record, but couldn't find anything. Should I implement this saving feature, and if so what would the structure of the save look like?

**Comment by karlzipser** [comment]
Right, I didn't implement this important function, have been doing it
manually so far. Also need to save state of training in order to be able to
resume with appropriate learning rate.

---------------------------------------
Karl Zipser, Ph.D.
Assistant Researcher
Helen Wills Neuroscience Institute,
Berkeley DeepDrive Center & Redwood Center for Theoretical Neuroscience
UC Berkeley
561 Evans hall, MC 3198
Berkeley, CA 94720-3198
. .... ...

**Comment by sauhaardac** [comment]
PyTorch Optimizers have state dictionaries just like the networks, these state dictionaries contain all relevant info for the individual optimizer (learning rate, momentum, etc.). In my old code I had this implemented, I will add this to the .weights save files, should only be a couple lines.

**Comment by karlzipser** [comment]
Put some comments into the code. Very clear and concise rewrite. I will try it out tomorrow. Big improvement.
